openapi/openapi.yaml

```diff
@@ -247,3 +250,209 @@ paths:
+              items:
+                $ref: '#/components/schemas/Namespace'
+          required:
+            - data
+
+    PageCount:
+      description: 'The number of items in this Page'
+      type: integer
+      example: 5
+
+    PageInfo:
+      description: 'Pagination info'
+      title: 'Page Info'
+      properties:
+        meta:
+          $ref: '#/components/schemas/PageMeta'
+        links:
+          $ref: '#/components/schemas/PageLinks'
+      required:
+        - meta
+        - links
+
+    PageLinks:
+      type: object
+      title: 'PageLinks'
+      properties:
+        first:
+          type: string
+          description: 'Link to first page'
+          format: uri
+          example: '/namespaces/?offset=0&limit=1'
+        last:
+          type: string
+          description: 'Link to last page'
+          format: uri
+          example: '/namespaces/?offset=7&limit=1'
+        next:
+          type: string
+          description: 'Link to next page'
+          format: uri
+          example: '/namespaces/?offset=5&limit=1'
+          nullable: true
+        previous:
+          type: string
+          description: 'Link to next page'
+          format: uri
+          example: '/namespaces/?offset=3&limit=1'
+          nullable: true
+      required:
+        - first
+        - last
+        - next
+        - previous
+
+    PageMeta:
+      title: 'PageMeta'
+      type: object
+      properties:
+        count:
+          $ref: '#/components/schemas/PageCount'
+      required:
+        - count
+
+    User:
+      title: 'User'
+      description: 'Automation Hub User'
+      type: object
+
+  parameters:
+
+    PageLimit:
+      description: 'Number of results to return per page.'
+      in: query
+      name: limit
+      required: false
+      schema:
+        type: integer
+        default: 10
+        minimum: 1
+        maximum: 100
+
+    PageOffset:
+      description: 'Page offset number within the paginated result set'
+      in: query
+      name: offset
+      required: false
+      schema:
+        type: integer
+        default: 0
+        minimum: 0
+
+    NamespaceName:
+      description: 'Namespace name'
+      in: path
+      name: name
+      required: true
+      schema:
+        type: string
+
+  requestBodies:
+
+    Namespace:
+      description: "A Namespace body"
+      content:
+        application/vnd.api+json:
+          schema:
+            $ref: '#/components/schemas/Namespace'
+
+  responses:
+
+    DeleteSuccessful:
+      description: 'A DELETE operation was successful'
+
+    Namespace:
+      description: 'Result of creating a Namespace'
+      content:
+        application/vnd.api+json:
+          schema:
+            $ref: '#/components/schemas/Namespace'
+
+    Namespaces:
+      description: 'Response containing a Page of a list of Namespaces'
+      content:
+        application/vnd.api+json:
+          schema:
+            $ref: '#/components/schemas/NamespacesPage'
```

Comment:
It seems you're extracting response specific schema to a standalone entry, while it's used only here for a particular response, which represents paginated list of Namespace objects.
I think it creates to many references and makes harder navigation in spec. Can you please avoid this.

Reply:
changed